src/encoder.rs

```diff
@@ -196,7 +195,47 @@ impl<'a> Encoder<'a> {
 
 
   #[inline]
-  fn write_int(&mut self, val: i64) -> CodecResult<()> {
+  fn write_int(&mut self, val: &PyObject) -> CodecResult<()> {
+    let size: u64 = val.call_method(self.py, "bit_length", NoArgs, None)?.extract(self.py)?;
+    let size: u32 = (size / 8 + 1) as u32;
+    if size <= 4 {
+      let v: i64 = FromPyObject::extract(self.py, val)?;
+      self.write_4byte_int(v)
+    } else {
+      self.write_arbitrary_int(val, size)
+    }
+  }
+
+  #[inline]
```

Comment:
When the function grows large it makes sense to remove `#[inline]`

Reply:
absolutley, is there any good guide lines for when to switch? There are other functions that are longer with inline.

Is it anything else then size when using inline?

Reply:
The difference is probably not visible at this scale when the function is called 1 or 2 times, but if we had something which is used dozens of times, then it would become a lot of code duplication. It just triggered in my head, it is too large, maybe ask to disable the inline.